src/peer/mod.rs

```diff
@@ -447,13 +450,20 @@ impl PeerConnection {
         {
             let id = self.id;
             let weak_sender = Rc::downgrade(&self.peer_events_sender);
+            let connections = Rc::downgrade(&self.connections);
             self.peer.on_connection_state_change(Some(
                 move |peer_connection_state| {
                     if let Some(sender) = weak_sender.upgrade() {
+                        let member_ids = connections
+                            .upgrade()
+                            .map(|connections| connections.member_ids())
+                            .unwrap_or_default();
```

Comment:
You probably should just return if connection weak ref upgrade failed. This would mean that both peer and peers repository was dropped already, so there is really no point in doing anything in this case.